pyproject.toml

```diff
@@ -24,6 +24,11 @@ statsmodels = {version = "^0.13.2", optional = true}
 dask = {version = "^2021.2.0", optional = true}
 distributed = {version = "^2021.2.0", optional = true}
 pandas = {version = "^1.2.4", optional = true}
+botorch = {version = "^0.7.1", optional = true}
+# version 1.13.0 has a bug where many unnecessary heavy dependencies are installed in
+# the base configuration of torch. Fixing it to <1.13 until the bug is resolved.
+# https://github.com/pytorch/pytorch/issues/88869
+torch = {version = "~1.12", optional = true}
```

Comment:
The background on this is that since `1.13` all kinds of cuda dependencies come with installing `torch` and I'm wondering if the way how to get a cpu only version of torch changed to installing from https://download.pytorch.org/whl/cpu; Do you know anything about this @sfalkner?

Reply:
I'll change the dependency to install from that extra index URL.

Reply:
I don't seem to be able to find a way to take care of this without incredibly slowing down poetry update. When I specify a new source:
```toml
[[tool.poetry.source]]
name = "torch-cpu"
url = "https://download.pytorch.org/whl/cpu/"
```
and then specify for torch to be fetched from that source, it'll still check the torch repo for all other dependencies before going to pypi.org; this is mitigated to some extent by also specifying pypi explicitly as a source and setting all first level dependencies to be fetched from there, but for some (I guess higher level dependencies) we still run into the issue that poetry first checks the torch index, slowing down the update by ~10s per dependency.

My proposal would be to just go for the default gpu enabled torch, accepting the downside that setup takes longer and requires more disk space for the botorch extra. Thoughts, @aeivazi & @sfalkner?

Reply:
Thanks for the suggestion, but without success I tried:
```toml
[[tool.poetry.source]]
name = "pypi_"
url = "https://pypi.python.org/simple"
default = true

[[tool.poetry.source]]
name = "torch-cpu"
url = "https://download.pytorch.org/whl/cpu/"
default = false
secondary = true

[tool.poetry.dependencies]
python = ">=3.8,<3.11"
parameterspace = "^0.7.2"
numpy = {version = "^1.22.0", optional = true, source = "pypi_"}
plotly = {version = "^5.10.0", optional = true, source = "pypi_"}
scipy = {version = "^1.6.0", optional = true, source = "pypi_"}
statsmodels = {version = "^0.13.2", optional = true, source = "pypi_"}
dask = {version = "^2021.2.0", optional = true, source = "pypi_"}
distributed = {version = "^2021.2.0", optional = true, source = "pypi_"}
pandas = {version = "^1.2.4", optional = true, source = "pypi_"}
botorch = {version = "^0.7.1", optional = true, source = "pypi_"}
torch = {version = "^1.11", optional = true, source = "torch-cpu"}


[tool.poetry.dev-dependencies]
black = {version = "^22.3.0", source = "pypi_"}
isort = {version = "^5.10.1", source = "pypi_"}
pytest = {version = "^7.2.0", source = "pypi_"}
mypy = {version = "^0.950", source = "pypi_"}
pytest-timeout = {version = "^2.1.0", source = "pypi_"}
pytest-cov = {version = "^3.0.0", source = "pypi_"}
mkdocs = {version = "^1.3.0", source = "pypi_"}
mkdocstrings = {version="^0.18.1", extras=["python-legacy"], source = "pypi_"}
mkdocs-material = {version = "^8.2.11", source = "pypi_"}
mkdocs-gen-files = {version = "^0.3.4", source = "pypi_"}
mkdocs-awesome-pages-plugin = {version = "^2.7.0", source = "pypi_"}
mike = {version = "^1.1.2", source = "pypi_"}
pylint = {version = "^2.13.8", source = "pypi_"}
pydocstyle = {version = "^6.1.1", source = "pypi_"}
pre-commit = {version = "^2.18.1", source = "pypi_"}
toml = {version = "^0.10.2", source = "pypi_"}
types-toml = {version = "^0.10.7", source = "pypi_"}
kaleido = {version = "0.2.1", source = "pypi_"} # pin hard, because 0.2.1post1 fails to install
```

Reply:
Is the package name `torch-cpu` or simply `torch`? The pip command from the official pytorch docs reads
```pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cpu```
which makes me wonder if the package name should simply be `torch`.

Reply:
The package name is `torch`, the source index name we gave in our config was `torch-cpu` but this was fully in our scope and has nothing to do with anything the torch maintainers offer.

Reply:
```
torch = {version = "^1.11", optional = true, source = "torch-cpu"}
```
is where the dependency `torch` is defined, and `torch-cpu` is just a local name to reference to the extra index `https://download.pytorch.org/whl/cpu/`.

Reply:
Thanks for clarifying and sorry for my confusion.